src/gui/guiEngine.cpp

```diff
@@ -105,15 +105,20 @@ void MenuMusicFetcher::fetchSounds(const std::string &name,
 		return;
 	m_fetched.insert(name);
 	std::string base;
-	base = porting::path_share + DIR_DELIM + "sounds";
-	dst_paths.insert(base + DIR_DELIM + name + ".ogg");
-	int i;
-	for(i=0; i<10; i++)
-		dst_paths.insert(base + DIR_DELIM + name + "."+itos(i)+".ogg");
-	base = porting::path_user + DIR_DELIM + "sounds";
-	dst_paths.insert(base + DIR_DELIM + name + ".ogg");
-	for(i=0; i<10; i++)
-		dst_paths.insert(base + DIR_DELIM + name + "."+itos(i)+".ogg");
+	std::vector<fs::DirListNode> list;
+	base = porting::path_share + DIR_DELIM;
+	//Reusable local function
+	auto add_paths = [&dst_paths](const std::string name, const std::string base = "") {
+		dst_paths.insert(base + name + ".ogg");
+		for (int i = 0; i < 10; i++)
+			dst_paths.insert(base + name + "." + itos(i) +
+					 ".ogg");
+	};
+	add_paths(name, base + "sounds" + DIR_DELIM);
+	add_paths(name);
+	base = porting::path_user + DIR_DELIM;
+	add_paths(name, base + "sounds" + DIR_DELIM);
+	add_paths(name);
```

Comment:
This has a duplicate `add_paths(name)`.  Also, this will try to sound files relative to the current directory even if they're just a filename.  You could check if the path is absolute and look for it directly if it is, or look for it relative to the user/share paths if it isn't.

Also, instead of reusing `base` use separate variable names, e.g. `share_prefix` and `user_prefix` (or just put it in the argument list without storing it in a separate variable since it's only used once).

Reply:
Removed the extra, updated the variable names. Hard to actually detect if a request is direct or not, it's very inexpensive for the main menu to search these folders later, and I see no value in developing a consistent and safe detection function (which would have to look for DIR_DELIM character presence in the string and likely other possible characters. Feel free to suggest one that is safe and short. 

Interestingly enough, if one provides only the "main_menu.ogg" the default mtpath/sounds/ folder is searched, otherwise the user must provide an absolute path. 

The original function merely loaded up the user and shared sounds folder, whereas I added the absolute path ability to search.

Reply:
You could do something like this:
```C++
// Allow full paths
if (name.find(DIR_DELIM) != std::string::npos) {
	add_paths(name);
} else {
	std::string share_prefix = porting::path_share + DIR_DELIM;
	add_paths(name, share_prefix + "sounds" + DIR_DELIM);
	std::string user_prefix = porting::path_user + DIR_DELIM;
	add_paths(name, user_prefix + "sounds" + DIR_DELIM);
}
```